scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt

```diff
@@ -42,23 +44,57 @@ internal fun findDatasetsAndConnectors(
           ?.filter { it.varType == PARAMETERS_DATASET_ID }
           ?.forEach { parameter ->
             val parameterValue = scenario.parametersValues?.find { it.parameterId == parameter.id }
-            if (parameterValue != null && parameterValue.value != "") {
-              addDatasetAndConnector(
-                  datasetService,
-                  connectorService,
-                  organizationId,
-                  parameterValue.value,
-                  datasets,
-                  connectors,
-              )
-            }
+            addParameterValue(
+                parameterValue,
+                datasetService,
+                connectorService,
+                organizationId,
+                datasets,
+                connectors,
+            )
           }
     }
   }
   return DatasetsConnectors(
       datasets = datasets.values.toList(), connectors = connectors.values.toList())
 }
 
+private fun addParameterValue(
+    parameterValue: ScenarioRunTemplateParameterValue?,
+    datasetService: DatasetApiService,
+    connectorService: ConnectorApiService,
+    organizationId: String,
+    datasets: MutableMap<String, Dataset>,
+    connectors: MutableMap<String, Connector>,
+) {
+  if (parameterValue != null && parameterValue.value != "") {
+    val value = parameterValue.value
+    val datasetIdList = getDatasetIdListFromValue(value)
+    datasetIdList.forEach {
+      addDatasetAndConnector(
+          datasetService,
+          connectorService,
+          organizationId,
+          it,
+          datasets,
+          connectors,
+      )
+    }
+  }
+}
+
+internal fun getDatasetIdListFromValue(paramValue: String): List<String> {
+  if (paramValue.startsWith("[")) {
+    if (!(paramValue.last().equals(']'))) {
+      throw CsmClientException("Malformed dataset id list: must start with [ and end with ]")
```

Comment:
I suggest you include the value of `paramValue` in the exception message, so clients can know exactly which value is malformed.